app/controllers/v0/sign_in_controller.rb

```diff
@@ -248,9 +249,44 @@ def validate_callback_params(code, state, error)
       raise SignIn::Errors::MalformedParamsError.new message: 'State is not defined' unless state
     end
 
-    def validate_token_params(code, grant_type)
-      raise SignIn::Errors::MalformedParamsError.new message: 'Code is not defined' unless code
+    def validate_token_params(code, service_account_assertion, grant_type)
       raise SignIn::Errors::MalformedParamsError.new message: 'Grant Type is not defined' unless grant_type
+
+      case grant_type
+      when SignIn::Constants::Auth::AUTH_CODE
+        raise SignIn::Errors::MalformedParamsError.new message: 'Code is not defined' unless code
+      when SignIn::Constants::Auth::JWT_BEARER
+        unless service_account_assertion
+          raise SignIn::Errors::MalformedParamsError.new message: 'Service Account Assertion is not defined'
+        end
+      else
+        raise SignIn::Errors::GrantTypeValueError.new message: 'Grant Type is not valid'
+      end
+    end
+
+    def generate_service_account_token(service_account_assertion, grant_type)
+      decoded_service_account_assertion =
+        SignIn::ServiceAccountValidator.new(service_account_assertion:, grant_type:).perform
+      service_account_access_token =
+        SignIn::ServiceAccountAccessTokenJwtEncoder.new(decoded_service_account_assertion:).perform
+
+      sign_in_logger.token_log('service_account token',
+                               service_account_access_token,
+                               { service_account_id: decoded_service_account_assertion.service_account_id })
+      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_TOKEN_SUCCESS)
+      { data: service_account_access_token }
```

Comment:
During my testing I noticed `{ service_account_access_token: }` is missing here.
<img width="662" alt="Screen Shot 2023-06-29 at 1 14 55 PM" src="https://github.com/department-of-veterans-affairs/vets-api/assets/10853209/be5965a2-a61f-4780-89b4-aad774c7df76">

Reply:
Also, should we just name it `access_token`?

Reply:
Trying to make very clear the distinction between regular access_tokens for users and these since they both live in SiS in the same `/token` route. So prefixing `service_account` seems like the most straightforward way to do that. I'm easing up on it on the Identity Dashboard side.